middlewares/user/getTargetUserIdByPMChannelId.js

```diff
@@ -0,0 +1,57 @@
+const {StreamChat} = require('stream-chat');
+const Getstream = require('../../vendor/getstream');
+
+/**
+ *
+ * @param {import('express').Request} req
+ * @param {import('express').Response} res
+ * @param {import('express').NextFunction} next
+ * @returns
+ */
+const getTargetUserIdByPMChannelIdMiddleware = async (req, res, next) => {
+  const {userId} = req || {};
+  const {channelId} = req?.query || {};
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The code is destructuring `userId` and `channelId` from `req` and `req.query` respectively without checking if `req` is defined. This could lead to a TypeError if `req` is undefined.
```diff
-  const {userId} = req || {};
-  const {channelId} = req?.query || {};
+  const userId = req ? req.userId : undefined;
+  const channelId = req && req.query ? req.query.channelId : undefined;
```



<!-- This is an auto-generated comment by OSS CodeRabbit -->